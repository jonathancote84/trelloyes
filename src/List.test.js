import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import Card from './Card'
import store from './STORE'
import renderer from 'react-test-renderer';



describe('List component', () =>{
    it('renders without crashing', () => {
        //first create the DOM element to render the component into 
        const div = document.createElement('div');

        //render the component, this is the actual test, if something goes wrong it will fail here 
        ReactDOM.render( <List
            key={store.lists[0].id}
            header={store.lists[0].header}
            cards={store.lists[0].cardIds.map(id => store.allCards[id])}
        />
            
    , div);

        //clean up code 
        ReactDOM.unmountComponentAtNode(div);

    });
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<List
                key={store.lists[0].id}
                header={store.lists[0].header}
                cards={store.lists[0].cardIds.map(id => store.allCards[id])}
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})
describe('Cards component', ()=>{
//the test case 
it('renders without crashing', () => {
    //first create the DOM element to render the component into 
    const div = document.createElement('div');

    //render the component, this is the actual test, if something goes wrong it will fail here 
    ReactDOM.render( <Card
        key='0' 
        title='secondCard'
        content='this is the actual test'
      />, div);

    //clean up code 
    ReactDOM.unmountComponentAtNode(div);

});
it('renders the UI as expected', () => {
    const tree = renderer
        .create( 
            <Card/>

        )
        .toJSON();
    expect(tree).toMatchSnapshot();
})

})