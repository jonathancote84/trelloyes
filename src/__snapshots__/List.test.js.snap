// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Twelth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
